### PR TITLE
Now we are using code format checks

### PR DESCRIPTION
**Pull request by insspb**
I found that `pycodestyle` was already in `requirements.dev`, but was not used at all. 

I did some work to make it usable, but I prefer the usage of flake8. Current code changes will work with both of them. `flake8` is based on `pycodestyle`, but can be extended by other tools.

This pull request to check new `travis` configuration and discuss this change. 

I think it is a good point to remove `pycodestyle` dependency in future releases and stay with `flake8`

My next thoughts to switch from `betamax` to `vcr.py` and `pytest`, as nose is outdated. What you think @facetoe?

### CODE REVIEW

File: zenpy/lib/endpoint.py [hunk 1]
```diff
@@ -293,7 +293,7 @@ def __call__(self, query=None, **kwargs):
                     value = '"{}"'.format(value)
                 renamed_kwargs.update({key + ':': '%s' % value})
 
-        search_query = ['%s%s' % (key, value) for (key, value) in renamed_kwargs.items()]
+        search_query = ['%s%s' % (k, v) for (k, v) in renamed_kwargs.items()]
```
Comment (insspb) [inline zenpy/lib/endpoint.py]: This rename was required by python 27 checks. It raised some strange pep8 error with value reassignment from line 293.

**Comment by insspb** [comment]
Fuh, done :) 
What we get at the end of all changes: 

- Any developer can run tests locally with simple `tox` command. tests will generate the same result as on Travis. This should exclude some "Checking Travis" pulls. 
- `pycodestyle` was removed from `requirements.dev` and replaced with `flake8`. I do not think we need to wait for something.
- Beautiful Soup (`bs4`) and `lxml` modules removed from `requirements.dev`, not used in the project, but generating errors with lxml 4.3.1+ in python 3.8. We do not need to do a workaround for this (Travis errors). 
- Project documentation generation tests added. But only in python 3+, as some dependencies not backported to 2.7
- One-two unused imports removed at all
- All "star" imports replaced with complete description. Required by `PEP8`
- `install_aliases` functions and import replaced with simple `int` at d24f006
- Some formatting to pas PEP8 tests.

**Comment by facetoe** [comment]
Hey thanks for the effort you put into this, but I'm not too keen to merge it TBH. Basically, this PR doesn't solve a problem that I have. I like pep8 as much as anyone, but it's more of a guide then a prescription IMO. Sometimes I like to format things differently to what pep8 would define, and that's OK.

`lxml` and `bs4` are both used in the project, so can't really remove them (used in tools).

Regarding `betamax`, `vcr.py` and `pytest`, the current setup is working fine. There is no reason to move to another library that I can see. The fact `nose` is outdated isn't causing any issues, and porting things over might.

So long story short, I'm going to leave things as they are until an issue arises that gives a reason for change.

**Comment by insspb** [comment]
- The idea of dependencies updates and tox implementation is to allow developers to test things locally before pull.
- I already have a repository with pytest in test. It 100% compatible with old tests and found some warnings related to future versions. https://github.com/insspb/zenpy/tree/tests
- `lxml` and `bs4`, ok, agree. Tools was out of my range. Will back it with version limit to exclude 3.8 builds errors in future prs
- About formatting. I do not like some pep8 things too. It is all in you configuration rage. This PR if a givving opportunity to do it. 

What ideas of this PR you like and will merge if redone?
